include/zephyr/linker/devicetree_regions.h

```diff
@@ -290,4 +332,78 @@
  */
 #define LINKER_DT_REGION_MPU(mpu_fn) _CHECK_APPLY_FN(_DT_COMPATIBLE, _EXPAND_MPU_FN, mpu_fn)
 
+/**
+ * @brief Generate MMU regions from the device tree nodes with both
+ *        'zephyr,memory-region' and 'zephyr,memory-region-mmu' attributes.
+ *
+ * Helper macro to apply an MMU_FN macro to all the memory regions declared
+ * using the 'zephyr,memory-region-mmu' property and the 'zephyr,memory-region'
+ * compatible.
+ *
+ * @p MMU_FN must take the form:
+ *
+ * @code{.c}
+ *   #define MMU_FN(name, base, size, attr) ...
+ * @endcode
+ *
+ * The 'name', 'base' and 'size' parameters are taken from the DT node.
+ *
+ * The 'zephyr,memory-region-mmu' string array property is passed as an extended token
+ * to the MMU_FN macro using the 'attr' parameter, mmu property is defined
+ * in architecture MMU header, for example: include/zephyr/arch/arm64/arm_mmu.h
+ *
+ *
+ * Example devicetree fragment:
+ *
+ *     / {
+ *             soc {
+ *                     ccm: ccm@30380000 {
+ *                         compatible = "nxp,imx-ccm";
+ *                         reg = <0x30380000 DT_SIZE_K(64)>;
+ *                         zephyr,memory-region = "CCM";
+ *                         zephyr,memory-region-mmu = "MT_DEVICE_nGnRnE", "MT_P_RW_U_NA", "MT_NS";
+ *                     };
+ *             };
+ *     };
```

Comment:
This between `@code{.dts}` and `@endcode`

While you are at it, can you fix lines 250-257 in the same file?

Reply:
Done.

Reply:
Done.